### PR TITLE
Update Haskell.nix rev

### PR DESCRIPTION
**Pull request by rvl**
Hackage updates should be fixed in Haskell.nix now — https://github.com/input-output-hk/haskell.nix/commit/83c85a0f2200fa63ed6122d66317069e5c7da1da

### CODE REVIEW

**Comment by roberth** [comment]
bors try

File: nix/project.nix [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
+  haskell-language-server-source = pkgs.fetchgit
+    (builtins.removeAttrs (builtins.fromJSON (builtins.readFile ./haskell-language-server.json)) ["date"]);
   defaults = {
     configuration.packages.ghc.flags.ghci = lib.mkForce true;
     configuration.packages.ghci.flags.ghci = lib.mkForce true;
     configuration.reinstallableLibGhc = true;
     # This fixes a performance issue, probably https://gitlab.haskell.org/ghc/ghc/issues/15524
     configuration.packages.ghcide.configureFlags = [ "--enable-executable-dynamic" ];
+
+    # fixme: how to override a haskell.nix option?
```
Comment (roberth) [inline nix/project.nix:13]: This is the correct way to define haskell.nix options, but previously I have noticed that they aren't used by the `stackProject` function.

**Comment by roberth** [comment]
bors try

**Comment by roberth** [comment]
bors try

File: nix/project.nix (hunk 1 above)
Comment (rvl) [reply nix/project.nix:13]: OK great. This configuration does seem to prevent at least one error when fetching the shake branch from git.

**Comment by rvl** [comment]
According to hercules-ci, the ghcide exe built successfully :tada:. But the tests don't pass. I also got that test failure locally.

**Comment by roberth** [comment]
That's a big improvement, thanks! I'll merge this into the update branch

**Pull request merged**